src/calendar_duration.rs

```diff
@@ -49,6 +49,58 @@ impl fmt::Debug for CalendarDuration {
     }
 }
 
+impl fmt::Display for CalendarDuration {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("P")?;
+        if self.is_zero() {
+            return f.write_str("0D");
+        }
+        let years = self.months / 12;
+        let months = self.months % 12;
+        if years > 0 {
+            f.write_fmt(format_args!("{}Y", years))?;
+        }
+        if months > 0 {
+            f.write_fmt(format_args!("{}M", months))?;
+        }
+        if self.days > 0 {
+            f.write_fmt(format_args!("{}D", self.days))?;
+        }
+        let nanos = self.nanos_component();
+        if self.seconds > 0 || nanos > 0 {
```

Comment:
Invert the condition to return early here, please.

Reply:
Yes, that is better 👍.